idxmgmt/supporter.go

```diff
@@ -63,6 +66,20 @@ type indexState struct {
 	isSet      atomic.Bool
 }
 
+// newDataStreamSelector returns an outil.Selector which routes events to
+// a data stream based on well-defined data_stream.* fields in events.
+func newDataStreamSelector() (outputs.IndexSelector, error) {
+	fmtstr, err := fmtstr.CompileEvent(datastreams.IndexFormat)
+	if err != nil {
+		return nil, err
+	}
+	expr, err := outil.FmtSelectorExpr(fmtstr, "", outil.SelectorLowerCase)
+	if err != nil {
+		return nil, err
+	}
+	return outil.MakeSelector(expr), nil
+}
```

Comment:
What if ES does not support data streams? Then the events would still be ingested in the new way, but there would not be any base templates or data stream setup. Since it is very unlikely someone turns this on at this stage (since other pieces are missing) we can ignore it for now, but might lead to problems in the future. When introducing ILM we added covered for ES versions not supporting ILM.

Reply:
I expect we won't document the new `apm-server.data_streams.enabled` config -- it's meant as an internal config to be switched on when using Fleet/Agent, which implies data streams are supported.

When using Fleet the server will be configured with minimal privileges, so checking for data stream support won't even be possible in that mode AFAIK.

Reply:
makes sense;